File: backend/src/main/java/br/gov/ufpr/convida/domain/Report.java
package br.gov.ufpr.convida.domain;

import java.io.Serializable;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
public class Report implements Serializable{

    /**
     *
     */
    private static final long serialVersionUID = 1L;

    @Id
    private String id;
    private String userId;
    private String userName;
    private String userLastName;
    private String report;
    private Boolean ignored;

    public Report(){}

    
    

    public String getId(){
        return this.id;
    }

    public void setId(String id){
        this.id = id;
    }

	public String getReport() {
        return this.report;
    }

    public void setReport(String report) {
        this.report = report;
    }

    public boolean isIgnored() {
        return this.ignored;
    }

    public boolean getIgnored() {
        return this.ignored;
    }

    public void setIgnored(Boolean ignored) {
        this.ignored = ignored;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getUserLastName() {
        return userLastName;
    }

    public void setUserLastName(String userLastName) {
        this.userLastName = userLastName;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        Report other = (Report) obj;
        if (id == null) {
            if (other.id != null)
                return false;
        } else if (!id.equals(other.id))
            return false;
        return true;
    }
  
  
}

File: frontend/convida/lib/app/shared/models/report.dart
import 'package:convida/app/shared/models/user.dart';

class Report {
  String id;
  String report;
  String userId;
  String userName;
  String userLastName;
  bool ignored;
  bool active;

  Report({this.userId, this.userLastName, this.userName, this.report, this.ignored, this.active});

  Report.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    report = json['report'];
    userId = json['userId'];
    userName = json['userName'];
    userLastName = json['userLastName'];
    ignored = json['ignored'];
    active = json['active'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['userId'] = this.userId;
    data['userName']= this.userName;
    data['userLastName'] = this.userLastName;
    data['report'] = this.report;
    data['ignored'] = this.ignored;
    data['active'] = this.active;
    return data;
  }
}